inflate.c

```diff
@@ -1120,6 +1120,8 @@ int32_t Z_EXPORT PREFIX(inflate)(PREFIX3(stream) *strm, int32_t flush) {
             state->mode = MEM;
             return Z_MEM_ERROR;
         }
+    } else if (INFLATE_NEED_CHECKSUM(strm) && !state->wsize && flush == Z_FINISH) {
```

Comment:
I am wondering if it should only be done if also `&& err == Z_BUF_ERROR` -- that is, only when `NEEDBITS()` bails early.

Reply:
The reason why I suggest it is: because the only test that fails is when CHECK bails early in NEEDBITS. When CHECK doesn't bail early I assume it will do the chksum in that case statement...?

Reply:
I think adding check `&& err == Z_BUF_ERROR` might be warranted, but we will need another test for that to be 100% sure... Returning `Z_BUF_ERROR` tells me the stream is truncated in some way and that itself should be enough to not silently ignore the case even if that means there is another bug somewhere else.

Reply:
To check the other scenario where it doesn't bail early, just don't send `-r 25` to minideflate from my PR.

Reply:
Sorry, was doing some traveling for the holiday.  On it, now.

Reply:
k, pushed update.

Reply:
FYI, I added the other check to the test cases in #1603.

Reply:
I think having both tests is better if just to catch potential regressions... We can't just test existing failures (before fixing them)...